lightning/src/chain/channelmonitor.rs

```diff
@@ -2743,22 +2743,47 @@ impl<Signer: EcdsaChannelSigner> ChannelMonitorImpl<Signer> {
 	fn closure_conf_target(&self) -> ConfirmationTarget {
 		// Treat the sweep as urgent as long as there is at least one HTLC which is pending on a
 		// valid commitment transaction.
+		let mut minimum_expiry = None;
+		let update_minimum_expiry = |local_minimum: Option<u32>, global_minimum: &mut Option<u32>| {
+			if let Some(expiry) = local_minimum {
+				*global_minimum = Some(global_minimum.map_or(expiry, |m| cmp::min(expiry, m)));
+			}
+		};
```

Comment:
Nit: I think the following refactor is more readable:

```suggestion
		let mut update_minimum_expiry = |local_minimum: Option<u32>| {
			if let Some(expiry) = local_minimum {
				minimum_expiry = Some(cmp::min(expiry, minimum_expiry.unwrap_or(expiry)));
			}
		};
```